core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -212,23 +215,22 @@ abstract class AbstractFetcherThread(name: String,
                     case e: FatalExitError => throw e
                     case e: Throwable =>
                       error(s"Error getting offset for partition $topicPartition to broker ${sourceBroker.id}", e)
-                      updatePartitionsWithError(topicPartition)
+                      partitionsWithError += topicPartition
                   }
                 case _ =>
                   if (isRunning.get) {
                     error(s"Error for partition $topicPartition to broker %${sourceBroker.id}:${partitionData.exception.get}")
-                    updatePartitionsWithError(topicPartition)
+                    partitionsWithError += topicPartition
                   }
               }
             })
         }
       }
     }
 
-    if (partitionsWithError.nonEmpty) {
+    if (partitionsWithError.nonEmpty)
       debug("handling partitions with error for %s".format(partitionsWithError))
```

Comment:
Would it make sense to move this log to `handlePartitionsWithErrors`?

Reply:
This is the only case where we log error for partitions with error. In all other cases we do not log error message for partitions with error. I am not sure how useful it is to have this log and whether we should do it for other cases.

Given that this message is at debug level, it is probably not used in most cases and it is not that important. I don't have a strong opinion on this. If you like, I can move this to `handlePartitionsWithErrors` in a follow-up patch.

Reply:
Moving into handlePartitionsWithErrors makes sense.